src/catchup/ApplyCheckpointWork.cpp

```diff
@@ -137,6 +140,13 @@ ApplyCheckpointWork::applyHistoryOfSingleLedger()
         return false;
     }
 
+    if (header.ledgerSeq > mLedgerRange.mLast)
```

Comment:
unless you have a bug, this should never happen

Reply:
Oh I see: that's because you're calling `applyHistoryOfSingleLedger` when coming back from a `wait for publish` even though you may be `done` (see my other comment)